### ISSUE TITLE
PEP 8 Inspection over-generalizes blank lines around functions

### ISSUE THREAD

**Issue by yoleg** [PyCQA/pep8#457]
PEP 8 only requires blank lines around "top-level function and class definitions" and "method definitions inside a class". However, pep8 marks any function without a line above it as failing the rule.
 
This is especially troublesome for functions that are inside other functions or methods, because they are often short and have no reason for a line above:
 
 ```python
def test():
    def success_callback():  # no need for blank line above
        print('Success! A wonderful thing has completed!')
    def error_callback():  # no need for blank line above
        print('Error! The entire program is crashing around us!')
    run_async_function(success_callback, error_callback)
```

**Comment by sigmavirus24** [comment]
> This is especially troublesome for functions that are inside other functions or methods, because they are often *[in my experience]* short and have no reason for a line above:

I added a clarifying phrase to your statement. If you ever venture into the OpenStack code bases there are nested functions which are anything but short. This rule helps that code base become more legible.

That said, the literal phrasing of the PEP (not its spirit which is not something I'm going to debate with anyone) means that this rule over steps its responsibilities. I'd be in favor of retaining the check but in a separate error code specifically for nested functions (if we can even easily make that distinction) such that you could ignore that code for your entire code base. Would that be a reasonable accommodation to you @yoleg?

**Comment by yoleg** [comment]
@sigmavirus24 Yes, that should work. Your clarifying phrase is correct. I myself write long nested functions and would appreciate the ability to turn this inspection on and off. It is only when using callback-based libraries that I find myself using dozens of very short closure functions. 
 
Side note 1: Perhaps this error is more closely associated with the PEP 8 rule "Use blank lines in functions, sparingly, to indicate logical sections." After all, (long) nested functions could be considered "logical sections".
 
Side note 2: The inspection only seems to notice blank lines missing *above* the nested function. It seems to me more logical to check for a blank line both above and below, as this looks much cleaner when separating long nested functions. 
 
Side note 3: It is actually my IDE (IntelliJ with Python/ PyCharm plugin) that is using this library in their inspections. They do allow me to disable specific PEP8 errors.

**Comment by sigmavirus24** [comment]
Re side note 2: See #400 as a related issue.
